nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java

```diff
@@ -0,0 +1,133 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.integtests.dremio;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.hc.client5.http.classic.methods.HttpGet;
+import org.apache.hc.client5.http.classic.methods.HttpPost;
+import org.apache.hc.client5.http.classic.methods.HttpUriRequestBase;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class DremioHelper {
+  private String token;
+  private String projectId;
+  private String baseUrl;
+  private String catalogName;
+
+  DremioHelper(String projectId, String token, String baseUrl, String catalogName) {
+    this.projectId = projectId;
+    this.token = token;
+    this.baseUrl = baseUrl;
+    this.catalogName = catalogName;
+  }
+
+  private String createPayload(String query) {
+    String payload =
+        format("{ \"sql\": \"%s\", \"context\": [\"%s\", \"db\"] }", query, catalogName);
+    return payload;
+  }
+
+  private String executeHttpRequest(HttpUriRequestBase req) {
+    req.addHeader("Authorization", "Bearer " + token);
+    req.addHeader("Content-Type", "application/json");
+    try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+      return httpClient.execute(req, response -> EntityUtils.toString(response.getEntity()));
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private String executeHttpPost(String url, String payload) {
+    HttpPost httpPost = new HttpPost(url);
+    httpPost.setEntity(new StringEntity(payload));
+    return executeHttpRequest(httpPost);
+  }
+
+  private String executeHttpGet(String url) {
+    HttpGet httpGet = new HttpGet(url);
+    return executeHttpRequest(httpGet);
+  }
+
+  private String waitForJobStatus(String jobId) {
+    String jobState = "RUNNING";
+    String url = baseUrl + "/v0/projects/" + projectId + "/job/" + jobId;
+    Set<String> finalJobStatesList = new HashSet<>(asList("COMPLETED", "FAILED", "CANCELED"));
+    while (!finalJobStatesList.contains(jobState)) {
+      String result = executeHttpGet(url);
+      JSONObject jObject = new JSONObject(result);
+      jobState = jObject.getString("jobState");
+    }
+    return jobState;
+  }
+
+  private List<List<Object>> parseQueryResult(String jobId) {
+
+    String url = baseUrl + "/v0/projects/" + projectId + "/job/" + jobId + "/results";
```

Comment:
Please add a link to the web page describing the API

Reply:
done